pydm/widgets/timeplot.py

```diff
@@ -116,29 +242,92 @@ def resetUpdatesAsynchronously(self):
+
+        Returns : float
+        -------
+            The timestamp of the most recent data point recorded into the data buffer.
+
+        """
         return self.data_buffer[0, -1]
 
 
 class PyDMTimePlot(BasePlot):
+    """
+    TimePlot now supports two mode:
+
+    1. The "classic" mode, in which the x-axis shows the timestamps, and move to the left
+    2. The new mode, in which the x-axis shows the negative time in seconds from the starting time, at which x = 0.
+
+    To maintain backward compatibility, the default drawing mode is in the classic, "plot_by_timestamps" mode, for both
+    TimePlotCurveItem and TimePlotCurve.
+    """
     SynchronousMode = 1
     AsynchronousMode = 2
 
-    def __init__(self, parent=None, init_y_channels=[], background='default'):
-        self._bottom_axis = TimeAxisItem('bottom')
-        self._left_axis = AxisItem('left')
-        super(PyDMTimePlot, self).__init__(
-            parent=parent,
-            background=background,
-            axisItems={'bottom': self._bottom_axis, 'left': self._left_axis})
-        self.plotItem.disableAutoRange(ViewBox.XAxis)
-        self.getViewBox().setMouseEnabled(x=False)
-        self._bufferSize = 1200
+    def __init__(self, parent=None, init_y_channels=[], plot_by_timestamps=True, background='default',
+                 plot_display=None):
+        """
+        Parameters
+        ----------
+
+        parent : Widget
+            The parent widget of the chart.
+        init_y_channels : list
+            A list of PV channels to plot when the graph is ready.
+         plot_by_timestamps : bool
+            True if the x-axis shows the timestamps as ticks, and moves automatically to the left. False if the the
+            x-axis starts at 0, and shows relative time in negative values to the left.
+        background : str
+            The color descriptor for the background of the graph
+        plot_display : Display
+            The PyDM display that contains the graph.
+        """
+        self.plot_by_timestamps = plot_by_timestamps
+
+        self._left_axis = AxisItem("left")
+        if plot_by_timestamps:
+            self._bottom_axis = TimeAxisItem('bottom')
+        else:
+            self.starting_epoch_time = time.time()
+            self._bottom_axis = AxisItem('bottom')
+
+        super(PyDMTimePlot, self).__init__(parent=parent, background=background,
+                                           axisItems={"bottom": self._bottom_axis, "left": self._left_axis},
+                                           plot_display=plot_display)
+
+        if self.plot_by_timestamps:
+            self.plotItem.disableAutoRange(ViewBox.XAxis)
+            self.getViewBox().setMouseEnabled(x=False)
+        else:
+            self.plotItem.setRange(xRange=[DEFAULT_X_MIN, 0], padding=0)
+            self.plotItem.setLimits(xMax=0)
+
+        self._bufferSize = MINIMUM_BUFFER_SIZE
```

Comment:
The default buffer size, which is what you are defining here, should definitely not be set to the minimum buffer size.  You actually want it to be pretty big, so that people making long-running plots don't unintentionally lose data.  I'd make a new `DEFAULT_BUFFER_SIZE` constant and make it something very large, like 10000.  Since we use a numpy array with dtype=float, that works out to (64 bits * 10000)/8 = 80 kilobytes of storage per curve.  Not exactly going to fill up the RAM on modern hardware.

Reply:
I'm setting the self._bufferSize to DEFAULT_BUFFER_SIZE, set at 10000. The MINIMUM_BUFFER_SIZE is set to be 2.